test/supavisor/client_handler/stats_test.exs

```diff
@@ -0,0 +1,133 @@
+defmodule Supavisor.ClientHandler.StatsTest do
+  use ExUnit.Case, async: false
+
+  @subject Supavisor.ClientHandler
+
+  @moduletag telemetry: true
+
+  setup :external_id
+
+  setup ctx do
+    :telemetry.attach(
+      {ctx.test, :client},
+      [:supavisor, :client, :network, :stat],
+      &__MODULE__.handle_event/4,
+      self()
+    )
+
+    :telemetry.attach(
+      {ctx.test, :db},
+      [:supavisor, :db, :network, :stat],
+      &__MODULE__.handle_event/4,
+      self()
+    )
+
+    on_exit(fn ->
+      :telemetry.detach({ctx.test, :client})
+      :telemetry.detach({ctx.test, :db})
+    end)
+  end
+
+  def handle_event([:supavisor, name, :network, :stat], measurement, meta, pid) do
+    send(pid, {:telemetry, {name, measurement, meta}})
+  end
+
+  setup ctx do
+    conn =
+      start_supervised!(
+        {Postgrex,
+         hostname: "localhost",
+         port: Application.fetch_env!(:supavisor, :proxy_port_transaction),
+         database: ctx.db,
+         username: ctx.user,
+         password: "postgres"}
+      )
+
+    {:ok, conn: conn}
+  end
+
+  describe "client network usage" do
+    test "increase on query", ctx do
+      external_id = ctx.external_id
+
+      assert {:ok, _} = Postgrex.query(ctx.conn, "SELECT 1", [])
+
+      assert_received {:telemetry,
+                       {:client, %{recv_oct: recv, send_oct: sent}, %{tenant: ^external_id}}}
+
+      assert recv > 0
+      assert sent > 0
+    end
+
+    # XXX: This should fail or not?
+    # test "increase on just auth", ctx do
+    #   external_id = ctx.external_id
+    #
+    #   assert_received {:telemetry,
+    #                    {:client, %{recv_oct: recv, send_oct: sent}, %{tenant: ^external_id}}}
+    #   assert recv > 0
+    #   assert sent > 0
+    # end
+
+    # TODO: For some reason it fails
+    test "do not not increase if other tenant is used", ctx do
+      external_id = ctx.external_id
+
+      {:ok, other} = external_id(ctx, "another")
+
+      other_conn =
+        start_supervised!(
+          {Postgrex,
+           hostname: "localhost",
+           port: Application.fetch_env!(:supavisor, :proxy_port_transaction),
+           database: other.db,
+           username: other.user,
+           password: "postgres"},
+          id: :postgrex_another
+        )
+
+      other_id = other.external_id
+
+      assert {:ok, _} = Postgrex.query(other_conn, "SELECT 1", [])
+
+      # assert_received {:telemetry, {:client, _, %{tenant: ^other_id}}}
+      refute_received {:telemetry, {:client, _, %{tenant: ^external_id}}}
+    end
```

Comment:
I am investigating failure of this test. It checks if querying data for one connection do not cause socket usage for another connection. But for whatever reason, it fails (which mean we have cross-reporting, or I have wrong assumptions about when metrics are sent).

Reply:
Maybe metrics for external_id is reporting, but it's reporting zero?

Reply:
I will check that out. But it is weird that there is anything sent at all.

Reply:
Found the problem. It was reporting the data for handshake there, I needed to clear the message queue before doing anything extra.